### PR TITLE
kvserver: run TestReplicaCircuitBreaker_ExemptRequests with leader leases

### PR DESCRIPTION
**Pull request by iskettaneh**
This commit makes TestReplicaCircuitBreaker_ExemptRequests run with both
epoch and leader leases.

References: https://github.com/cockroachdb/cockroach/issues/133763

Release note: None

### CODE REVIEW

**Review by arulajmani — APPROVED** [review]
Nice work here. :LGTM: modulo the question in-line -- lemme know if you want me to have another look. 

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136412#-:-OCtcZqPBmPyOHB7mT5v:b-w5va7e)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @iskettaneh)

___
*[`pkg/kv/kvserver/storeliveness/support_manager.go` line 290 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/136412#-OCtbmsd1hOU-UHNI8K6:-OCtbmsd1hOU-UHNI8K7:b-8skv9x) ([raw file](https://github.com/cockroachdb/cockroach/blob/96572ade58fe39dcba250cedc8371e588d51e31c/pkg/kv/kvserver/storeliveness/support_manager.go#L290)):*
> ```Go
> 		return
> 	}
> 	if sm.knobs != nil && sm.knobs.DisableAllHeartbeats != nil && sm.knobs.DisableAllHeartbeats.Load() == true {
> ```

nit: `&& sm.knobs.DisableAllHeartbeats.Load() {`
___
*[`pkg/kv/kvserver/client_replica_circuit_breaker_test.go` line 1004 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136412#-OCtc7lGESPxYwomMaXZ:-OCtc7lGESPxYwomMaX_:b-oruwoj) ([raw file](https://github.com/cockroachdb/cockroach/blob/111ad92fb60ad1640f4d397c0f64ae6f29cdf365/pkg/kv/kvserver/client_replica_circuit_breaker_test.go#L1004)):*
> ```Go
> 
> 	if leaseType == roachpb.LeaseNone {
> 		// If no lease type was explicitly provided, let metamorphism choose between
> ```

Before merging, let's confirm that all these tests pass if leader leases are metamorphically enabled (instead of epoch based leases). (if you haven't already)

**Review by iskettaneh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136412)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani)

___
*[`pkg/kv/kvserver/client_replica_circuit_breaker_test.go` line 1004 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136412#-OCtc7lGESPxYwomMaXZ:-OCuCY6yE2UvsFQVNUJc:b5wgu9h) ([raw file](https://github.com/cockroachdb/cockroach/blob/111ad92fb60ad1640f4d397c0f64ae6f29cdf365/pkg/kv/kvserver/client_replica_circuit_breaker_test.go#L1004)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Before merging, let's confirm that all these tests pass if leader leases are metamorphically enabled (instead of epoch based leases). (if you haven't already)
</blockquote></details>

Yes I checked that

**Comment by iskettaneh** [comment]
TFTR!

bors r+

**Pull request merged**
